### DISCUSSION TITLE
Adding new product from newegg

### DISCUSSION THREAD

**Discussion by alrowa7i** [jef/streetmerchant#912] [Help]
### Description
What is the best way to add a new product from newegg to the bot?

**Comment by TyrantRex** [comment]
> ### Description
> What is the best way to add a new product from newegg to the bot?

These are the files you'll be modifying:
```
1. /streetmerchant/.env
2. /streetmerchant/src/store/model/store.ts
3. /streetmerchant/src/store/model/newegg.ts
```

**1. /streetmerchant/.env**
Modify the two lines below:
```
SHOW_ONLY_BRANDS="[yourbrandhere]"
SHOW_ONLY_MODELS="[yourmodelhere]"
```

**2. /streetmerchant/src/store/model/store.ts**

Under "export type Brand =" add..
```
| '[yourbrandhere]'
```
Under "export type Model =" add..
```
| '[yourmodelhere]'
```

**3. /streetmerchant/src/store/model/newegg.ts**
Scroll down to the links and add the product you want. I'm going to use a random 3080 + psu combo deal as an example for the carturl, itemnumber, series, and url.
```
{
    brand: '[yourbrandhere]',
    cartUrl:
        'https://secure.newegg.com/Shopping/AddtoCart.aspx?Submit=ADD&ItemList=Combo.4190361',
    itemNumber: 'Combo.4190361', #you can find this in the url
    model: '[yourmodelhere]',
    series: '3080', #optional but some examples are 3070, 3080
    url:
        'https://www.newegg.com/Product/ComboDealDetails?ItemList=Combo.4190361'
},
```

**Comment by rodalpho** [comment]
Thank you for the tutorial! Was a bit tricky but I managed to get it to work. Hope the devs can add the combos natively soon as I guess I can't update from source now. 

Don't forget to increase the max price in the .env file, anyone else doing this! Also I believe the hashed comments aren't valid, they will need to be removed also.

**Comment by howdoicomputer** [comment]
Something to note is that you don't need a custom brand. You can use the same brand of the card in the bundle. I only customized the model and just appended a b<number> on the end of it to indicate the bundle number.

It wouldn't be hard to draft a PR for this. Though... it'd probably be a good idea to add support for using an environment variable to toggle querying for bundles.

**Comment by Z474AL3MN2VC** [comment]
I just added the Powercolor Fighter to my NewEgg store. I'm realizing a lot of other entries do not have an itemNumber property. Is this required? I see it's used for the real time tracking method. I'm going to try and read through the code but I'm hoping items without the itemNumber don't fail and show "Out of Stock" when it's not really checking.

**Reply by Z474AL3MN2VC** [reply]
After setting up duplicate entries in my NewEgg store, one with and one without itemNumber, it looks like it's not required.

**Comment by lozbinks** [comment]
Hi guys, I've just tried adding to new products (the xbox wireless headset and ps5 charging station) by following the instructions provided by  @TyrantRex however when I go to run I'm getting 2 errors..

1: src/store/lookup.ts:442:22 - error TS7053: Element implicitly has an 'any' type because expression of type 'Series' can't be used to index type '

2: src/notification/discord.ts:56:42 - error TS7053: Element implicitly has an 'any' type because expression of type 'Series' can't be used to index type '

Does anyone know how to resolve these/what needs correcting?

Many thanks

Update:

Figured it out. 

You also need to edit the config file with the new series max price and notifications section.
